test/activate-event.html

```diff
@@ -49,8 +49,13 @@
 
       var s;
 
-      setup(function () {
+      setup(function(done) {
         s = fixture('test');
+
+        s.addEventListener('iron-items-changed', function onIronItemsChanged() {
```

Comment:
Oh, so that didn't actually explain the fix... So the most reliable fix seemed to be to *only* update items and selection when the `observeNodes` callback is called, which means that the tests need to wait for `iron-items-changed` to be sure that `items` and other things that depend on `items` have been updated.